osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+            "flush_time_per_shard": self.aggregate_json_by_key("flush_time_per_shard"),
+            "flush_count": self.aggregate_json_by_key("flush_count"),
+            "merge_throttle_time": self.aggregate_json_by_key("merge_throttle_time"),
+            "merge_throttle_time_per_shard": self.aggregate_json_by_key("merge_throttle_time_per_shard"),
+            "ml_processing_time": self.aggregate_json_by_key("ml_processing_time"),
+            "young_gc_time": self.aggregate_json_by_key("young_gc_time"),
+            "young_gc_count": self.aggregate_json_by_key("young_gc_count"),
+            "old_gc_time": self.aggregate_json_by_key("old_gc_time"),
+            "old_gc_count": self.aggregate_json_by_key("old_gc_count"),
+            "memory_segments": self.aggregate_json_by_key("memory_segments"),
+            "memory_doc_values": self.aggregate_json_by_key("memory_doc_values"),
+            "memory_terms": self.aggregate_json_by_key("memory_terms"),
+            "memory_norms": self.aggregate_json_by_key("memory_norms"),
+            "memory_points": self.aggregate_json_by_key("memory_points"),
+            "memory_stored_fields": self.aggregate_json_by_key("memory_stored_fields"),
+            "store_size": self.aggregate_json_by_key("store_size"),
+            "translog_size": self.aggregate_json_by_key("translog_size"),
+            "segment_count": self.aggregate_json_by_key("segment_count"),
+            "total_transform_search_times": self.aggregate_json_by_key("total_transform_search_times"),
+            "total_transform_index_times": self.aggregate_json_by_key("total_transform_index_times"),
+            "total_transform_processing_times": self.aggregate_json_by_key("total_transform_processing_times"),
+            "total_transform_throughput": self.aggregate_json_by_key("total_transform_throughput")
+        }
+
+        for task, task_metrics in self.accumulated_results.items():
+            iterations = self.accumulated_iterations.get(task, 1)
+            aggregated_task_metrics = self.calculate_weighted_average(task_metrics, iterations)
+            op_metric = {
+                "task": task,
+                "operation": task,
+                "throughput": aggregated_task_metrics["throughput"],
+                "latency": aggregated_task_metrics["latency"],
+                "service_time": aggregated_task_metrics["service_time"],
+                "client_processing_time": aggregated_task_metrics["client_processing_time"],
+                "processing_time": aggregated_task_metrics["processing_time"],
+                "error_rate": aggregated_task_metrics["error_rate"],
+                "duration": aggregated_task_metrics["duration"]
+            }
+            aggregated_results["op-metrics"].append(op_metric)
+
+        # extract the necessary data from the first test execution, since the configurations should be identical for all test executions
+        test_exe_store = metrics.test_execution_store(self.config)
```

Comment:
This is the same as `test_store` right? To avoid initializing several test execution stores, would it be better to just initialize `metrics.test_execution_store(self.config)` as an instance variable when creating Aggregator instance?

Reply:
You're right, I did this a couple times. Cleaned it up